test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2336,4 +2336,23 @@ public static <T extends Throwable> T expectThrows(Class<T> expectedType, Reques
             () -> builder.get().decRef() // dec ref if we unexpectedly fail to not leak transport response
         );
     }
+
+    /**
+     * Call a supplier that might throw and fail if it throws, useful to avoid try/catch boilerplate or cumbersome
+     * propagation of checked exceptions around something you assume should never throw.
+     *
+     * @param throwingSupplier A supplier that may throw an exception
+     * @return The value returned by the supplier
+     */
+    protected static <T> T assertDoesNotThrow(ThrowingSupplier<T> throwingSupplier) {
+        try {
+            return throwingSupplier.get();
+        } catch (Exception e) {
+            return fail(e);
+        }
+    }
+
+    protected interface ThrowingSupplier<T> {
+        T get() throws Exception;
+    }
```

Comment:
This might be controversial. As the comment suggests it's just to reduce boilerplate in places where we don't want checked exceptions to propagate, and we don't expect them to occur. There's something like this in JUnit 5. I kept the definitions local because I'm not trying to introduce a grand new pattern here, but rather a useful convenience given the new behaviour of `PlainActionFuture#result`.